### PR TITLE
LPS-170830 Created right-click actions according to "Abandoned" and "Complete" status

### PR DESCRIPTION
**Pull request by rafaelfs98**
[LPS-170515](https://issues.liferay.com/browse/LPS-170515)

### CODE REVIEW

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Testflow/useTestflowActions.tsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+		{
+			action: (task) =>
+				navigate(
+					`/project/${task.build?.project?.id}/routines/${task.build?.routine?.id}/build/${task.build?.id}`
+				),
+			icon: 'page',
+			name: i18n.translate('view-associete-build'),
+		},
+
+		{
```
Comment (samuelleolopes) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Testflow/useTestflowActions.tsx:40]: Remove unnecessary line

```suggestion
		},
		{
```

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Testflow/useTestflowActions.tsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+						}
+					);
+				}),
+			hidden: ({dueStatus}) =>
+				dueStatus?.key === TaskStatuses.IN_ANALYSIS,
+			icon: 'polls',
+			name: i18n.translate('reanalyze'),
+		},
+
+		{
```
Comment (samuelleolopes) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Testflow/useTestflowActions.tsx:58]: Remove unnecessary line

```suggestion
		},
		{
```

**Review by caiosouza15 — APPROVED** [review]
Great work :fire:

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/i18n/en_US.ts [hunk 3]
```diff
@@ -213,6 +213,8 @@ export default {
 	'mark-at-least-one-case-to-export': 'Mark at least one case to export.',
 	'mark-at-least-one-case-type-for-processing':
 		'Mark at least one case type for processing.',
+	'mark-at-least-one-user-for-assignment':
+		'Mark At Least One User For Assignment.',
```
Comment (welitonpe) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/i18n/en_US.ts:217]: Do all words really need to be capitalized?

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Testflow/useTestflowActions.tsx [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+			action: (task, mutate) =>
+				testrayTaskImpl.reanalyze(task).then(() => {
+					updateItemFromList(
+						mutate,
+						0,
+						{},
+						{
+							revalidate: true,
+						}
+					);
```
Comment (welitonpe) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Testflow/useTestflowActions.tsx:49]: Simplify returning directly

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Testflow/useTestflowActions.tsx [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+			action: (subtask, mutate) =>
+				testrayTaskImpl.reanalyze(subtask).then(() => {
+					updateItemFromList(
+						mutate,
+						0,
+						{},
+						{
+							revalidate: true,
+						}
+					);
```
Comment (welitonpe) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Testflow/useTestflowActions.tsx:66]: Simplify returning directly

**Review by welitonpe — COMMENTED** [review]
Good Job :fire: :fire:

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Testflow/TaskHeaderActions.tsx [hunk 6]
```diff
@@ -57,14 +58,11 @@ const TaskHeaderActions = () => {
...
-				.then((response) => {
-					mutateTask(response);
-
-					revalidateTaskUser();
-				}),
+		onSave: async (newUserIds: number[]) => {
+			await testrayTaskUsersImpl.assign(testrayTask.id, newUserIds);
+
+			revalidateTaskUser();
+		},
```
Comment (kevenleone) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Testflow/TaskHeaderActions.tsx:65]: This is not wrong, but we discussed about the possibility of newUserIds be equal to 0 and remove all the assignees.

It should be at least 1 (throw an alert, the same used in TestflowForm.tsx)

File: modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Testflow/modal/index.tsx [hunk 7]
```diff
@@ -40,7 +41,16 @@ const TestflowAssignUserModal: React.FC<TestflowAssignUserModalProps> = ({
...
-					onSubmit={() => onSave(state)}
+					onSubmit={() =>
+						state.length
+							? onSave(state)
+							: Liferay.Util.openToast({
+									message: i18n.translate(
+										'mark-at-least-one-user-for-assignment'
+									),
+									type: 'danger',
+							  })
```
Comment (kevenleone) [inline modules/apps/site-initializer/site-initializer-testray/site-initializer-testray/extra/remote-app/src/pages/Testflow/modal/index.tsx:52]: I am not sure if this validation should be here, probably where it's called, during the onSave according to my last comment

**Review by kevenleone — CHANGES_REQUESTED** [review]
Good job @rafaelfs98 just some things to change

**Comment by rafaelfs98** [comment]
@kevenleone, @welitonpe  Suggestions accepted and implemented, thanks for the suggestions were very constructive. You can now verify the changes in the new commit

**Review by kevenleone — APPROVED** [review]
LGMT 🔥

**Comment by kevenleone** [comment]
ci:test:sf

**Comment by kevenleone** [comment]
ci:forward

**Comment by kevenleone** [comment]
ci:forward

**Comment by kevenleone** [comment]
ci:forward

**Pull request closed without merging**